lib/collection/src/shards/local_shard/mod.rs

```diff
@@ -969,13 +970,13 @@ impl LocalShard {
         vector_size * info.points_count
     }
 
-    pub async fn local_shard_info(&self) -> CollectionInfoInternal {
+    pub async fn local_shard_info(&self) -> ShardInfoInternal {
```

Comment:
Should I just create a dedicated function for deciding shard status:
```rs
pub async fn local_shard_status(&self) -> ShardStatus { ... }
```

It won't have to compute `vectors_count`, `schema`, etc. However, it will have the part of the code from `local_shard_info` and will have to be kept in sync.

Reply:
```rs
pub async fn local_shard_status(&self) -> ShardStatus {
        let segments = self.segments().read();
        let mut status = ShardStatus::Green;
        let mut optimizer_status = OptimizersStatus::Ok;

        for (idx, segment) in segments.iter() {
            if segment_info.segment_type == SegmentType::Special {
                status = ShardStatus::Yellow;
            }

            if !segments.failed_operation.is_empty() || segments.optimizer_errors.is_some() {
                status = ShardStatus::Red;
            }

            if let Some(error) = &segments.optimizer_errors {
                optimizer_status = OptimizersStatus::Error(error.to_string());
            }
        }

        // If still green while optimization conditions are triggered, mark as grey
        if status == ShardStatus::Green
            && self.update_handler.lock().await.has_pending_optimizations()
            && optimizer_status == OptimizersStatus::Ok
        {
            status = ShardStatus::Grey;
            // optimizer_status =
            //     OptimizersStatus::Error("optimizations pending, awaiting update operation".into());
        }

        status
}
```

Reply:
separate function is preferred, as it looks like it would allow us to keep `get_telemetry_data` sync

Reply:
I would actually vote to just keep `local_shard_info` without a separate function for the status. The reason is that there is quite a bit of overlap and we'll most likely fail to keep it properly in sync.

The only overhead I see in the current function is some number additions. Those are negledable.

Reply:
On second thought, I have just discussed this with @generall: 

Let's do separate functions. But the `local_shard_info` we already have will simply call `local_shard_status` to get the color. That way we don't have duplication. There is a bit of overhead, but it's good enough here.